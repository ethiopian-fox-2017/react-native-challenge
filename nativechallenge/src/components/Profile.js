import React, { Component } from 'react';
import { Container,
  Content,
  ListItem,
  Text,
  Button,
  ActionSheet,
  Header,
  Left,
  Body,
  Right,
  Title,
  Icon } from 'native-base';

var BUTTONS = [
  'Option 0',
  'Option 1',
  'Option 2',
  'Delete',
  'Cancel',
];

var DESTRUCTIVE_INDEX = 3;
var CANCEL_INDEX = 4;

class Profile extends Component {
    constructor(props) {
      super(props)

    }


    render() {
        return (
            <Container>
                <Header>
                  <Left>
                    <Button transparent>
                      <Icon name="heart" size={20} color='red' />
                    </Button>
                  </Left>
                  <Body>
                    <Title>Hello</Title>
                  </Body>
                  <Right />
                </Header>
                <Content>

                    <Button onPress={()=> ActionSheet.show(
                      {
                        options: BUTTONS,
                        cancelButtonIndex: CANCEL_INDEX,
                        destructiveButtonIndex: DESTRUCTIVE_INDEX,
                        title: 'Testing ActionSheet'
                      },
                      (buttonIndex) => {
                        this.setState({ clicked: BUTTONS[buttonIndex] });
                      }
                      )}><Text>Actionsheet</Text></Button>
                    <ListItem itemHeader first>
                        <Text>COMEDY</Text>
                    </ListItem>
                    <ListItem >
                        <Text>Hangover</Text>
                    </ListItem>
                    <ListItem>
                        <Text>Horrible Bosses</Text>
                    </ListItem>
                    <ListItem last>
                        <Text>Conjuring</Text>
                    </ListItem>
                    <ListItem itemHeader>
                        <Text>ACTION</Text>
                    </ListItem>
                    <ListItem>
                        <Text>Terminator Genesis</Text>
                    </ListItem>
                </Content>
            </Container>
        );
    }
}

export default Profile
