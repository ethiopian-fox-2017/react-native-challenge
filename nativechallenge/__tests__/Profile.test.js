import 'react-native';
import React from 'react';
import Profile from '../src/components/Profile.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

it('renders correctly', () => {
  const tree = shallow(
    <Profile />
  );
  expect(tree).toMatchSnapshot()
});
