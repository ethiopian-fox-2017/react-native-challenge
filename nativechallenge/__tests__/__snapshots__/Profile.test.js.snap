// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(Container)
    virtual={undefined}
>
    <Styled(Header)
        virtual={undefined}
    >
        <Styled(Left)
            virtual={undefined}
        >
            <Styled(Button)
                transparent={true}
                virtual={undefined}
            >
                <Styled(Icon)
                    color="red"
                    name="heart"
                    size={20}
                    virtual={undefined}
                />
            </Styled(Button)>
        </Styled(Left)>
        <Styled(Body)
            virtual={undefined}
        >
            <Styled(Title)
                virtual={undefined}
            >
                Hello
            </Styled(Title)>
        </Styled(Body)>
        <Styled(Right)
            virtual={undefined}
        />
    </Styled(Header)>
    <Styled(Content)
        virtual={undefined}
    >
        <Styled(Button)
            onPress={[Function]}
            virtual={undefined}
        >
            <Styled(Text)
                virtual={undefined}
            >
                Actionsheet
            </Styled(Text)>
        </Styled(Button)>
        <Styled(ListItem)
            first={true}
            itemHeader={true}
            virtual={undefined}
        >
            <Styled(Text)
                virtual={undefined}
            >
                COMEDY
            </Styled(Text)>
        </Styled(ListItem)>
        <Styled(ListItem)
            virtual={undefined}
        >
            <Styled(Text)
                virtual={undefined}
            >
                Hangover
            </Styled(Text)>
        </Styled(ListItem)>
        <Styled(ListItem)
            virtual={undefined}
        >
            <Styled(Text)
                virtual={undefined}
            >
                Horrible Bosses
            </Styled(Text)>
        </Styled(ListItem)>
        <Styled(ListItem)
            last={true}
            virtual={undefined}
        >
            <Styled(Text)
                virtual={undefined}
            >
                Conjuring
            </Styled(Text)>
        </Styled(ListItem)>
        <Styled(ListItem)
            itemHeader={true}
            virtual={undefined}
        >
            <Styled(Text)
                virtual={undefined}
            >
                ACTION
            </Styled(Text)>
        </Styled(ListItem)>
        <Styled(ListItem)
            virtual={undefined}
        >
            <Styled(Text)
                virtual={undefined}
            >
                Terminator Genesis
            </Styled(Text)>
        </Styled(ListItem)>
    </Styled(Content)>
</Styled(Container)>,
  "nodes": Array [
    <Styled(Container)
      virtual={undefined}
>
      <Styled(Header)
            virtual={undefined}
      >
            <Styled(Left)
                  virtual={undefined}
            >
                  <Styled(Button)
                        transparent={true}
                        virtual={undefined}
                  >
                        <Styled(Icon)
                              color="red"
                              name="heart"
                              size={20}
                              virtual={undefined}
                        />
                  </Styled(Button)>
            </Styled(Left)>
            <Styled(Body)
                  virtual={undefined}
            >
                  <Styled(Title)
                        virtual={undefined}
                  >
                        Hello
                  </Styled(Title)>
            </Styled(Body)>
            <Styled(Right)
                  virtual={undefined}
            />
      </Styled(Header)>
      <Styled(Content)
            virtual={undefined}
      >
            <Styled(Button)
                  onPress={[Function]}
                  virtual={undefined}
            >
                  <Styled(Text)
                        virtual={undefined}
                  >
                        Actionsheet
                  </Styled(Text)>
            </Styled(Button)>
            <Styled(ListItem)
                  first={true}
                  itemHeader={true}
                  virtual={undefined}
            >
                  <Styled(Text)
                        virtual={undefined}
                  >
                        COMEDY
                  </Styled(Text)>
            </Styled(ListItem)>
            <Styled(ListItem)
                  virtual={undefined}
            >
                  <Styled(Text)
                        virtual={undefined}
                  >
                        Hangover
                  </Styled(Text)>
            </Styled(ListItem)>
            <Styled(ListItem)
                  virtual={undefined}
            >
                  <Styled(Text)
                        virtual={undefined}
                  >
                        Horrible Bosses
                  </Styled(Text)>
            </Styled(ListItem)>
            <Styled(ListItem)
                  last={true}
                  virtual={undefined}
            >
                  <Styled(Text)
                        virtual={undefined}
                  >
                        Conjuring
                  </Styled(Text)>
            </Styled(ListItem)>
            <Styled(ListItem)
                  itemHeader={true}
                  virtual={undefined}
            >
                  <Styled(Text)
                        virtual={undefined}
                  >
                        ACTION
                  </Styled(Text)>
            </Styled(ListItem)>
            <Styled(ListItem)
                  virtual={undefined}
            >
                  <Styled(Text)
                        virtual={undefined}
                  >
                        Terminator Genesis
                  </Styled(Text)>
            </Styled(ListItem)>
      </Styled(Content)>
</Styled(Container)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Profile />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Profile {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(Container)
          virtual={undefined}
>
          <Styled(Header)
                    virtual={undefined}
          >
                    <Styled(Left)
                              virtual={undefined}
                    >
                              <Styled(Button)
                                        transparent={true}
                                        virtual={undefined}
                              >
                                        <Styled(Icon)
                                                  color="red"
                                                  name="heart"
                                                  size={20}
                                                  virtual={undefined}
                                        />
                              </Styled(Button)>
                    </Styled(Left)>
                    <Styled(Body)
                              virtual={undefined}
                    >
                              <Styled(Title)
                                        virtual={undefined}
                              >
                                        Hello
                              </Styled(Title)>
                    </Styled(Body)>
                    <Styled(Right)
                              virtual={undefined}
                    />
          </Styled(Header)>
          <Styled(Content)
                    virtual={undefined}
          >
                    <Styled(Button)
                              onPress={[Function]}
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        Actionsheet
                              </Styled(Text)>
                    </Styled(Button)>
                    <Styled(ListItem)
                              first={true}
                              itemHeader={true}
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        COMEDY
                              </Styled(Text)>
                    </Styled(ListItem)>
                    <Styled(ListItem)
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        Hangover
                              </Styled(Text)>
                    </Styled(ListItem)>
                    <Styled(ListItem)
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        Horrible Bosses
                              </Styled(Text)>
                    </Styled(ListItem)>
                    <Styled(ListItem)
                              last={true}
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        Conjuring
                              </Styled(Text)>
                    </Styled(ListItem)>
                    <Styled(ListItem)
                              itemHeader={true}
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        ACTION
                              </Styled(Text)>
                    </Styled(ListItem)>
                    <Styled(ListItem)
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        Terminator Genesis
                              </Styled(Text)>
                    </Styled(ListItem)>
          </Styled(Content)>
</Styled(Container)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(Container)
          virtual={undefined}
>
          <Styled(Header)
                    virtual={undefined}
          >
                    <Styled(Left)
                              virtual={undefined}
                    >
                              <Styled(Button)
                                        transparent={true}
                                        virtual={undefined}
                              >
                                        <Styled(Icon)
                                                  color="red"
                                                  name="heart"
                                                  size={20}
                                                  virtual={undefined}
                                        />
                              </Styled(Button)>
                    </Styled(Left)>
                    <Styled(Body)
                              virtual={undefined}
                    >
                              <Styled(Title)
                                        virtual={undefined}
                              >
                                        Hello
                              </Styled(Title)>
                    </Styled(Body)>
                    <Styled(Right)
                              virtual={undefined}
                    />
          </Styled(Header)>
          <Styled(Content)
                    virtual={undefined}
          >
                    <Styled(Button)
                              onPress={[Function]}
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        Actionsheet
                              </Styled(Text)>
                    </Styled(Button)>
                    <Styled(ListItem)
                              first={true}
                              itemHeader={true}
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        COMEDY
                              </Styled(Text)>
                    </Styled(ListItem)>
                    <Styled(ListItem)
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        Hangover
                              </Styled(Text)>
                    </Styled(ListItem)>
                    <Styled(ListItem)
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        Horrible Bosses
                              </Styled(Text)>
                    </Styled(ListItem)>
                    <Styled(ListItem)
                              last={true}
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        Conjuring
                              </Styled(Text)>
                    </Styled(ListItem)>
                    <Styled(ListItem)
                              itemHeader={true}
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        ACTION
                              </Styled(Text)>
                    </Styled(ListItem)>
                    <Styled(ListItem)
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              >
                                        Terminator Genesis
                              </Styled(Text)>
                    </Styled(ListItem)>
          </Styled(Content)>
</Styled(Container)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Profile />,
}
`;
